src/Listener.cpp

```diff
@@ -143,9 +142,37 @@ std::string Listener::ThreadID()  {
 bool Listener::NotificationReceived() {
    std::vector<std::string> dataFromQueue;
    mQueueReader->GetShot(getShotTimeout, dataFromQueue);
-   return MessageHasPayload(dataFromQueue);
+   bool notificationReceived = MessageHasPayload(dataFromQueue);
+   if (notificationReceived) {
```

Comment:
clear the previous messages that were stored. Independent whether the new message should be  stored or not.,